sdk/storage/storage-blob/src/BlobClient.ts

```diff
@@ -1593,24 +1593,30 @@ export class BlobClient extends StorageClient {
     // "https://myaccount.blob.core.windows.net/mycontainer/blob?sasString";
     // "https://myaccount.blob.core.windows.net/mycontainer/blob";
 
-    let urlWithoutSAS = this.url.split("?")[0]; // removing the sas part of url if present
-    urlWithoutSAS = urlWithoutSAS.endsWith("/") ? urlWithoutSAS.slice(0, -1) : urlWithoutSAS; // Slicing off '/' at the end if exists
+    try {
```

Comment:
I didn't consider "/" in a blobName while writing this parser.
Let me verify "/" in a blobname and update the code if needed.

Reply:
Yes we can do a bug-fix if there's an issue.

---
In reply to: [331716712](https://github.com/Azure/azure-sdk-for-js/pull/5405#discussion_r331716712) [](ancestors = 331716712)

Reply:
https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata#blob-names

Reply:
Fixed.